pkg/controller/elasticsearch/shutdown/node_test.go

```diff
@@ -0,0 +1,385 @@
+	type args struct {
+		typ    esclient.ShutdownType
+		status *esclient.ShutdownStatus
+	}
+	tests := []struct {
+		name       string
+		args       args
+		fixture    string
+		wantErr    bool
+		wantDelete bool
+	}{
+		{
+			name:    "Respect type when deleting shutdowns",
+			fixture: shutdownFixture,
+			args: args{
+				typ:    esclient.Restart,
+				status: &esclient.ShutdownComplete,
+			},
+			wantErr:    false,
+			wantDelete: false,
+		},
+		{
+			name:    "Respect status when deleting shutdowns",
+			fixture: shutdownFixture,
+			args: args{
+				typ:    esclient.Remove,
+				status: &esclient.ShutdownStarted,
+			},
+			wantErr:    false,
+			wantDelete: false,
+		},
+		{
+			name:    "Allow all status values when deleting shutdowns",
+			fixture: shutdownFixture,
+			args: args{
+				typ:    esclient.Remove,
+				status: nil,
+			},
+			wantErr:    false,
+			wantDelete: true,
+		},
+		{
+			name:    "Should delete shutdowns",
+			fixture: shutdownFixture,
+			args: args{
+				typ:    esclient.Remove,
+				status: &esclient.ShutdownComplete,
+			},
+			wantErr:    false,
+			wantDelete: true,
+		},
+		{
+			name: "Should bubble up errors",
+			args: args{
+				typ:    esclient.Remove,
+				status: &esclient.ShutdownComplete,
+			},
+			fixture:    `{not json`,
+			wantErr:    true,
+			wantDelete: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			deleteCalled := false
+			client := esclient.NewMockClient(version.MustParse("7.14.0"), func(req *http.Request) *http.Response {
+				if req.Method == http.MethodDelete {
+					deleteCalled = true
+				}
+				return &http.Response{
+					StatusCode: 200,
+					Body:       ioutil.NopCloser(bytes.NewBuffer([]byte(tt.fixture))),
+					Header:     make(http.Header),
+					Request:    req,
+				}
+			})
+			ns := &NodeShutdown{
+				c:   client,
+				typ: tt.args.typ,
+				log: log.Log.WithName("test"),
+			}
+
+			if err := ns.Clear(context.Background(), tt.args.status); (err != nil) != tt.wantErr {
+				t.Errorf("Clear() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if tt.wantDelete != deleteCalled {
+				t.Errorf("Clear() deletedCalled = %v, wantDelete %v", deleteCalled, tt.wantDelete)
+			}
+		})
+	}
+}
+
+func TestNodeShutdown_ShutdownStatus(t *testing.T) {
+	type args struct {
+		podToNodeID map[string]string
+		podName     string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		fixture string
+		want    NodeShutdownStatus
+		wantErr bool
+	}{
+		{
+			name: "unknown node",
+			args: args{
+				podToNodeID: nil,
+				podName:     "pod-1",
+			},
+			want:    NodeShutdownStatus{},
+			wantErr: true,
+		},
+		{
+			name:    "successful lookup",
+			fixture: shutdownFixture,
+			args: args{
+				podToNodeID: map[string]string{
+					"pod-1": "txXw-Kd2Q6K0PbYMAPzH-Q",
+				},
+				podName: "pod-1",
+			},
+			want: NodeShutdownStatus{
+				Status:      esclient.ShutdownComplete,
+				Explanation: "",
```

Comment:
nit: maybe use an API response which includes an explanation if it is expected to have it reported in `NodeShutdownStatus`:

for example:

```golang
var (
	stalledFixture = `{
  "nodes" : [
    {
      "node_id" : "MT7eFuG5QliYII9K2HCgLw",
      "type" : "REMOVE",
      "reason" : "93475",
      "shutdown_startedmillis" : 1637057097778,
      "status" : "STALLED",
      "shard_migration" : {
        "status" : "STALLED",
        "shard_migrations_remaining" : 41,
        "explanation" : "shard [0] [primary] of index [.geoip_databases] cannot move, use the Cluster Allocation Explain API on this shard for details"
      },
      "persistent_tasks" : {
        "status" : "COMPLETE"
      },
      "plugins" : {
        "status" : "COMPLETE"
      }
    },
    {
      "node_id" : "rX6VerB4RUuZVzGSb8uSIw",
      "type" : "REMOVE",
      "reason" : "93475",
      "shutdown_startedmillis" : 1637057097175,
      "status" : "STALLED",
      "shard_migration" : {
        "status" : "STALLED",
        "shard_migrations_remaining" : 26,
        "explanation" : "shard [0] [primary] of index [.ds-metrics-kubernetes.apiserver-default-2021.11.16-000001] cannot move, use the Cluster Allocation Explain API on this shard for details"
      },
      "persistent_tasks" : {
        "status" : "COMPLETE"
      },
      "plugins" : {
        "status" : "COMPLETE"
      }
    }
  ]
}`
)
...
		{
			name:    "successful lookup with explanation",
			fixture: stalledFixture,
			args: args{
				podToNodeID: map[string]string{
					"pod-1": "MT7eFuG5QliYII9K2HCgLw",
				},
				podName: "pod-1",
			},
			want: NodeShutdownStatus{
				Status:      esclient.ShutdownStalled,
				Explanation: "shard [0] [primary] of index [.geoip_databases] cannot move, use the Cluster Allocation Explain API on this shard for details",
			},
			wantErr: false,
		},
```